source/base/function_spherical.cc

```diff
@@ -0,0 +1,341 @@
+  (subst(srules, ratcoeff(expand(H3), ur.utheta))) - c_utheta_ur;
+  c_utheta_uphi : (subst(srules, ratcoeff(expand(H3), utheta.uphi)));
+  (subst(srules, ratcoeff(expand(H3), uphi.utheta))) - c_utheta_uphi;
+  c_ur2 : (subst(srules, ratcoeff(expand(H3), ur.ur)));
+  c_ur_uphi : (subst(srules, ratcoeff(expand(H3), ur.uphi)));
+  (subst(srules, ratcoeff(expand(H3), uphi.ur))) - c_ur_uphi;
+  c_uphi2 : (subst(srules, ratcoeff(expand(H3), uphi.uphi)));
+
+
+  where (used later to do tensor products):
+
+  ur     : [cos(theta)*sin(phi), sin(theta)*sin(phi), cos(phi)];
+  utheta : [-sin(theta), cos(theta), 0];
+  uphi   : [cos(theta)*cos(phi), sin(theta)*cos(phi), -sin(phi)];
+
+  with the following proof of substitution rules above:
+
+  -diff(ur,theta)+sin(phi)*utheta;
+  trigsimp(-diff(utheta,theta)-cos(phi)*uphi-sin(phi)*ur);
+  -diff(uphi,theta)+cos(phi)*utheta;
+  -diff(ur,phi)+uphi;
+  -diff(uphi,phi)-ur;
+
+   */
+
+  namespace
+  {
+
+    /**
+     * Evaluate unit vectors in spherical coordinates
+     */
+    template <int dim>
+    void set_unit_vectors(const double &cos_theta,
+                          const double &sin_theta,
+                          const double &cos_phi,
+                          const double &sin_phi,
+                          Tensor<1,dim> &unit_r,
+                          Tensor<1,dim> &unit_theta,
+                          Tensor<1,dim> &unit_phi)
+    {
+      unit_r[0]     = cos_theta * sin_phi;
+      unit_r[1]     = sin_theta * sin_phi;
+      unit_r[2]     = cos_phi;
+
+      unit_theta[0] = -sin_theta;
+      unit_theta[1] =  cos_theta;
+      unit_theta[2] = 0.;
+
+      unit_phi[0]   = cos_theta * cos_phi;
+      unit_phi[1]   = sin_theta * cos_phi;
+      unit_phi[2]   = -sin_phi;
+    }
+
+
+    /**
+     * calculates out[i][j] += v*(in1[i]*in2[j]+in1[j]*in2[i])
+     */
+    template <int dim>
+    void add_outer_product(SymmetricTensor<2,dim> &out,
+                           const double &val,
+                           const Tensor<1,dim> &in1,
+                           const Tensor<1,dim> &in2)
+    {
+      if (val != 0.)
+        for (unsigned int i = 0; i < dim; i++)
+          for (unsigned int j = i; j < dim; j++)
+            out[i][j] += (in1[i]*in2[j]+in1[j]*in2[i])*val;
+    }
+
+    /**
+     * calculates out[i][j] = v*in[i]in[j]
+     */
+    template <int dim>
+    void add_outer_product(SymmetricTensor<2,dim> &out,
+                           const double &val,
+                           const Tensor<1,dim> &in)
+    {
+      if (val != 0.)
+        for (unsigned int i = 0; i < dim; i++)
+          for (unsigned int j = i; j < dim; j++)
+            out[i][j] += val*in[i]*in[j];
+    }
+  }
+
+  template <int dim>
+  Spherical<dim>::Spherical(const Point<dim> p)
+    :
+    Function<dim>(1),
+    coordinate_system_offset(p)
+  {
+    AssertThrow(dim==3,
+                ExcNotImplemented());
+  }
+
+  template <int dim>
+  Spherical<dim>::~Spherical()
+  {
+  }
```

Comment:
You don't actually need the destructor. It is automatically virtual because the one in the base class already is.